clang/lib/StaticAnalyzer/Checkers/MallocChecker.cpp

```diff
@@ -1877,6 +1914,33 @@ static bool didPreviousFreeFail(ProgramStateRef State,
   return false;
 }
 
+static void printOwnershipTakesList(raw_ostream &os, CheckerContext &C,
+                                    const Expr *E) {
+  if (const CallExpr *CE = dyn_cast<CallExpr>(E)) {
+    const FunctionDecl *FD = CE->getDirectCallee();
+    if (!FD)
+      return;
+
+    if (FD->hasAttrs()) {
```

Comment:
You could switch this into an early-return to reduce indentation for the rest of the code.
Same applies to the `if (const CallExpr *CE = dyn_cast<CallExpr>(E))` a few lines above.